### PR TITLE
Bring back missing tests

### PR DESCRIPTION
**Pull request by Leouarz**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- [ ] Feature
- [ ] Bugfix
- [ ] Refactor
- [ ] Format
- [ ] Documentation
- [x] Testing
- [ ] Other:

## Description

Add more tests for filters
Closes https://github.com/availproject/engineering/issues/559
Closes https://github.com/availproject/engineering/issues/560

## Checklist
- [x] I have performed a self-review of my own code.
- [x] The tests pass successfully with `cargo test`.
- [x] The code was formatted with `cargo fmt`.
- [x] The code compiles with no new warnings with `cargo build --release` and `cargo build --release --features runtime-benchmarks`.
- [x] The code has no new warnings when using `cargo clippy`.
- [x] If this change affects documented features or needs new documentation, I have created a PR with a [documentation update](https://github.com/availproject/availproject.github.io).

### CODE REVIEW

File: base/src/calls_proof.rs
Comment (0xSasaPrsic) [inline base/src/calls_proof.rs:1]: Why is this deleted/moved? We should update docs that point to the removed files.

Comment (Leouarz) [reply base/src/calls_proof.rs:1]: What doc included something about calls_proof ? I don't see it used anywhere

**Review by 0xSasaPrsic — APPROVED** [review]

**Review by aterentic-ethernal — APPROVED** [review]
